compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -1452,6 +1456,46 @@ public void visit(BLangDynamicArgExpr dynamicParamExpr) {
         result = dynamicParamExpr;
     }
 
+    @Override
+    public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
+        resolveClosuresInInterpolations(regExpTemplateLiteral.reDisjunction.sequenceList);
+        result = regExpTemplateLiteral;
+    }
+
+    private void resolveClosuresInInterpolations(List<BLangExpression> sequenceList) {
+        for (BLangExpression seq : sequenceList) {
+            if (seq.getKind() != NodeKind.REG_EXP_SEQUENCE) {
+                return;
+            }
+            BLangReSequence sequence = (BLangReSequence) seq;
+            for (BLangExpression term : sequence.termList) {
+                if (term.getKind() != NodeKind.REG_EXP_ATOM_QUANTIFIER) {
+                    continue;
+                }
+                BLangExpression atom = ((BLangReAtomQuantifier) term).atom;
+                NodeKind kind = atom.getKind();
+                if (!isReAtomNode(kind)) {
+                    atom = rewriteExpr(atom);
+                    continue;
+                }
+                if (kind == NodeKind.REG_EXP_CAPTURING_GROUP) {
+                    resolveClosuresInInterpolations(((BLangReCapturingGroups) atom).disjunction.sequenceList);
+                }
+            }
+        }
+    }
+
+    private boolean isReAtomNode(NodeKind kind) {
```

Comment:
This been duplicated in couple of places, check whether we can move it to a util?